projects/exui-common-lib/src/lib/models/address-option.model.ts

```diff
@@ -0,0 +1,29 @@
+import { AddressModel } from '.';
+
+export class AddressOption {
+  public description: string;
+  public value: AddressModel;
+
+  constructor (addressModel: AddressModel, description: string) {
+    if (description === null) {
+      this.value = addressModel;
+      this.description = this.getDescription();
+    } else {
+      this.description = description;
+    }
+  }
+
+  private getDescription() {
+    return this.removeInitialCommaIfPresent(
+      `${this.value.addressLine1 === undefined ? '' : this.value.addressLine1}${this.prefixWithCommaIfPresent(this.value.addressLine2)}${this.prefixWithCommaIfPresent(this.value.addressLine3)}, ${this.value.postTown}`
+    );
+  }
+
+  private prefixWithCommaIfPresent(value: string) {
```

Comment:
@johnbenjamin-hmcts - missing return type

Reply:
@Daniel-Lam - I have updated the code.